client/lib/query-manager/index.js

```diff
@@ -104,6 +104,27 @@ export default class QueryManager {
 	}
 
 	/**
+	 * A sorting function that defines the sort order of items under
+	 * consideration of the specified query.
+	 *
+	 * @param  {Array}  keys  Keys to be sorted
+	 * @param  {Array}  items Items by which to sort
+	 * @param  {Object} query Query object
+	 * @return {Array}        Sorted keys
+	 */
+	sort( keys, items, query ) {
+		return keys.sort( ( keyA, keyB ) => {
+			if ( ! items[ keyA ] || ! items[ keyB ] ) {
```

Comment:
This is purely a nitpick because I've got nothing better to do right now but this could be simplified to

```js
    if ( items[ keyA ] && items[ keyB ] ) {
        // None of the items has been removed so sort em
        return this.compare( query, items[ keyA ], items[ keyB ] );
    }

    return 0;
```